packages/client-cli/test/frontend-openapi.test.mjs

```diff
@@ -35,11 +35,20 @@ test('build basic client from url', async (t) => {
   match(types, /interface GetRedirectRequest/)
   match(types, /interface GetRedirectResponseFound/)
   match(types, /interface GetRedirectResponseBadRequest/)
-  match(types, /'message': string \| null;/)
-  match(types, /'dateTime': string \| Date;/)
-  match(types, /'otherDate': string \| Date;/)
-  match(types, /'nullableDate': string \| Date \| null;/)
-  match(types, /'normalString': string;/)
+
+  // Request can contain a `Date` type
+  ok(types.includes("'messageReq': string | null;"))
+  ok(types.includes("'dateTimeReq': string | Date;"))
+  ok(types.includes("'otherDateReq': string | Date;"))
+  ok(types.includes("'nullableDateReq': string | Date | null;"))
+  ok(types.includes("'normalStringReq': string;"))
+
+  // Response shouldn't contain `Date` for the same fields as above
+  ok(types.includes("'messageRes': string | null;"))
+  ok(types.includes("'dateTimeRes': string;"))
+  ok(types.includes("'otherDateRes': string;"))
+  ok(types.includes("'nullableDateRes': string | null;"))
+  ok(types.includes("'normalStringRes': string;"))
```

Comment:
why did you move away from `match`?

Reply:
Because with the previous regex, it didn't check the part after the `|` symbol.
So basically tests were passing (even though they should've failed).

Reply:
Why did you drop the regexp?

Reply:
As explained above, the previous regex didn't check the part after the `|` symbol.

Reply:
I might have been clearer: why did you not change the regex?

Reply:
I'm not a huge fan of regex (unless when they are essential), and this way the code seems more readable to me, but I can update if needed 😊

Reply:
Considering `|` is a special symbol for regex, here a more complex expression should be created. I don't think it's worth it to be honest, but I can do it 😉

Reply:
`match` provides a slightly better error message, and that's why it was used here.
Thanks

Reply:
Wait, `match` doesn't work.
If I add `match(true, false)` the test pass.

I think it's due to the fact the instead of throwing an error it returns a boolean 🤔

Reply:
If I add `strictEqual(true, false)` from the `node:test` it fails as it should

Reply:
So I think the `match` function should be updated (and check if any regression has been added meanwhile). Meanwhile I propose to merge the current PR as it is 😉

Reply:
All match here became no-ops :(.